File: laboratorios/lab02/l-javascript/clases/Reporte.js
import { Profesor } from "./Profesor.js"
import { Estudiante } from "./Estudiante.js";

export class Reporte {
  profesor

  constructor(profesor) {
    this.profesor = profesor instanceof Profesor ? profesor : null
  }

  // Dependencia con estudiante
  generarReporte(estudiante) {
    if (!(estudiante instanceof Estudiante)) {
      return "Reporte no disponible"
    }
    let reporte = `Reporte del Profesor: ${this.profesor.nombre}\n`
    reporte += `, acerca del Estudiante: ${estudiante.nombre}`
    return reporte
  }

  toString() {
    return `Reporte{profesor=${this.profesor}}`
  }
}

File: laboratorios/lab02/l-java/src/main/java/com/tuorg/poo/clases/Reporte.java
package com.tuorg.poo.clases;

public class Reporte {
    Profesor profesor;

    public Reporte(Profesor profesor) {
        this.profesor = profesor;
    }

    public Profesor getProfesor() {
        return profesor;
    }
    public void setProfesor(Profesor profesor) {
        this.profesor = profesor;
    }

    //Dependencia con Estudiante
    public void generarReporte(Estudiante estudiante) {
        System.out.println("Reporte del Profesor: " + profesor.getNombre() + ", acerca del Estudiante: " + estudiante.getNombre());
    }

    @Override
    public String toString() {
        return "Reporte{" +
                "profesor=" + profesor +
                '}';
    }
}
